### PR TITLE
APP-7007: implement DisableAuthService CLI command

### PR DESCRIPTION
**Pull request by gloriacai01**
[Ticket](https://viam.atlassian.net/browse/APP-7007):
Screenshot 2025-01-07 at 2 31 14 PM
Manual testing: 
abort option
Screenshot 2025-01-08 at 1 30 34 PM
disable option (didn't have an org with auth service enabled yet, but the rpc response is expected)
Screenshot 2025-01-08 at 6 06 38 PM

### CODE REVIEW

File: cli/client.go [hunk 1]
```diff
@@ -190,6 +190,68 @@ func (c *viamClient) organizationsSupportEmailGetAction(cCtx *cli.Context, orgID
...
+// DisableAuthServiceConfirmation is the Before action for 'organizations auth-service disable'.
+// It asks for the user to confirm that they want to disable the auth service.
+func DisableAuthServiceConfirmation(c *cli.Context, args disableAuthServiceArgs) error {
+	if args.OrgID == "" {
+		return errors.New("cannot disable auth service without an organization ID")
+	}
+
+	yellow := "\033[1;33m%s\033[0m"
+	printf(c.App.Writer, yellow, "WARNING!!\n")
+	printf(c.App.Writer, yellow, fmt.Sprintf("You are trying to disable auth service for organization ID %s. "+
```
Comment (maxhorowitz) [inline cli/client.go:206]:
```suggestion
	printf(c.App.Writer, yellow, fmt.Sprintf("You are trying to disable the auth service for organization ID %s. "+
```
Sorry for nit - but I think this reads better

File: cli/client.go [hunk 2]
```diff
@@ -190,6 +190,68 @@ func (c *viamClient) organizationsSupportEmailGetAction(cCtx *cli.Context, orgID
...
+		"Once disabled, this will remove and permanently delete all white-labeled login flows for any applications "+
+		"associated with your organization ID\n", args.OrgID))
```
Comment (maxhorowitz) [inline cli/client.go:208]:
```suggestion
		"Once disabled, all custom auth views and emails will be removed from your organization's (%s) " + " 
		"OAuth applications and permanently deleted.\n", args.OrgID))
```

Comment (maxhorowitz) [reply cli/client.go:208]: Want to noodle on this some more but just a suggestion. The only behavioral difference here is that there I want to be clear that it removes more than just the login screen. It really removes any view associated with authN + any emails (forgot password email + email verification email are 2 examples).

File: cli/client.go [hunk 3]
```diff
@@ -190,6 +190,68 @@ func (c *viamClient) organizationsSupportEmailGetAction(cCtx *cli.Context, orgID
...
+	printf(c.App.Writer, yellow, "If you wish to continue, please type \"disable\":")
```
Comment (maxhorowitz) [inline cli/client.go:209]: 👍 Great

File: cli/client.go [hunk 4]
```diff
@@ -190,6 +190,68 @@ func (c *viamClient) organizationsSupportEmailGetAction(cCtx *cli.Context, orgID
...
+		return err
+	}
+
+	rawInput, err := bufio.NewReader(c.App.Reader).ReadString('\n')
+	if err != nil {
+		return err
+	}
+
+	input := strings.ToUpper(strings.TrimSpace(rawInput))
+	if input != "DISABLE" {
```
Comment (maxhorowitz) [inline cli/client.go:220]: Sorry for the super nit !
```suggestion
	if input := strings.ToUpper(strings.TrimSpace(rawInput)); input != "DISABLE" {
```

File: cli/client.go [hunk 5]
```diff
@@ -190,6 +190,68 @@ func (c *viamClient) organizationsSupportEmailGetAction(cCtx *cli.Context, orgID
...
+	if err != nil {
+		return err
+	}
+
+	orgID := args.OrgID
+	if orgID == "" {
+		return errors.New("cannot disable auth service without an organization ID")
+	}
+
+	return c.disableAuthServiceAction(cCtx, args.OrgID)
```
Comment (maxhorowitz) [inline cli/client.go:238]:
```suggestion
	return c.disableAuthServiceAction(cCtx, orgID)
```
(only suggesting because you already pulled out above)

File: cli/client_test.go [hunk 6]
```diff
@@ -371,6 +371,25 @@ func TestEnableAuthServiceAction(t *testing.T) {
 	test.That(t, out.messages[0], test.ShouldContainSubstring, "enabled auth")
 }
 
+func TestDisableAuthServiceAction(t *testing.T) {
+	disableAuthServiceFunc := func(ctx context.Context, in *apppb.DisableAuthServiceRequest, opts ...grpc.CallOption) (
```
Comment (maxhorowitz) [inline cli/client_test.go:375]: Any way to validate
1. the empty org ID
2. the valid org ID followed by an abort
3. the valid org ID followed by a DISABLE

? If not easy lmk - but would hope to assert that stuff in code in addition to our manual tests

**Review by maxhorowitz — CHANGES_REQUESTED** [review]
Mostly nits (sorry!) - one Q about testing coverage - and one Q about removal of `auth-service`

**Comment by purplenicole730** [comment]
> Mostly nits (sorry!) - one Q about testing coverage - and one Q about removal of `auth-service`

If you can, can you add all these comments into one review so that it's easier to follow, and it doesn't spam others' emails 😅 ?

File: cli/app.go [hunk 7]
```diff
@@ -474,12 +474,20 @@ var app = &cli.App{
 							},
 							Action: createCommandWithT[enableAuthServiceArgs](EnableAuthServiceAction),
 						},
-					},
-				},
-				{
-					Name:  "auth-service",
```
[the remark this answers is not in the document]
Comment (purplenicole730) [reply cli/app.go:480]: Agreed. Otherwise, this becomes a breaking change

File: cli/client.go [hunk 8]
```diff
@@ -190,6 +190,68 @@ func (c *viamClient) organizationsSupportEmailGetAction(cCtx *cli.Context, orgID
...
+func DisableAuthServiceConfirmation(c *cli.Context, args disableAuthServiceArgs) error {
+	if args.OrgID == "" {
+		return errors.New("cannot disable auth service without an organization ID")
+	}
+
+	yellow := "\033[1;33m%s\033[0m"
+	printf(c.App.Writer, yellow, "WARNING!!\n")
+	printf(c.App.Writer, yellow, fmt.Sprintf("You are trying to disable auth service for organization ID %s. "+
+		"Once disabled, all custom auth views and emails will be removed from your organization's (%s) " + " 
+		"OAuth applications and permanently deleted.\n", args.OrgID))
```
Comment (purplenicole730) [inline cli/client.go:208]: The image in the PR description doesn't seem to match this text. Please update. I'd like to see it since I'm not sure what the second `%s` value is.

**Review by purplenicole730 — CHANGES_REQUESTED** [review]

File: cli/client_test.go (hunk 6 above)
Comment (gloriacai01) [reply cli/client_test.go:375]: i dont know if there is an easy way to cases 2 and 3 here, @purplenicole730 do u know how i could test this? but i did add a test case for the empty org ID case

Comment (purplenicole730) [reply cli/client_test.go:375]: I'm not too sure, since I'm not too familiar with the Before capability (which I'm assuming is what `DISABLE` is from). My best guess is to see how the other functions that have Before capabilities are tested.

Comment (purplenicole730) [reply cli/client_test.go:375]: If you can't figure it out after searching on the internet/our code, then at least include this test in the description so that we know you manually tested it.

File: cli/app.go [hunk 9]
```diff
@@ -491,12 +491,20 @@ var app = &cli.App{
...
-					Usage: "manage auth-service",
-					Subcommands: []*cli.Command{
+						{
+							Name:      "disable",
+							Usage:     "disable auth-service for OAuth applications",
+							UsageText: createUsageText("disable", []string{generalFlagOrgID}, true),
```
Comment (purplenicole730) [inline cli/app.go:497]: I just realized, why is `otheroptions` marked as true? The only option is the required one, so this should be set to false.

File: cli/client_test.go (hunk 6 above)
Comment (maxhorowitz) [reply cli/client_test.go:375]: @purplenicole730 SGTM !

File: cli/app.go (hunk 9 above)
Comment (maxhorowitz) [reply cli/app.go:497]: Great catch

**Review by maxhorowitz — APPROVED** [review]
Action items before merge:
1. Address Nicole's comment about `otheroptions`
2. Add comment describing manual test if not unit testable

Otherwise, great work!

File: cli/client_test.go (hunk 6 above)
Comment (gloriacai01) [reply cli/client_test.go:375]: I couldn't find any existing tests for the Before action , but I added my manual testing in the description!

**Review by purplenicole730 — APPROVED** [review]
